stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -192,6 +197,51 @@ public class MessageComposerController(
                 }.launchIn(scope)
             }
         }
+
+        setupComposerState()
+    }
+
+    /**
+     * Sets up the observing operations for various composer states.
+     */
+    private fun setupComposerState() {
+        scope.launch {
```

Comment:
As discussed, the outer coroutine launch here is redundant.

Reply:
Good catch! What about combining all the flows here?

Reply:
Combining flows is terrible, we'd have to do something like:

```
flow1.combine(flow2) {1, 2 -> pair }
.combine(flow3) {(1,2), 3 -> Triple }
...
```

And so on for 9 properties. This seemed like a nicer solution honestly, we could also not expose all the individual properties if possible and just update the MessageComposerState directly, that could make it shorter and easier.

Reply:
Up to several flows we have convenient methods:
```
public fun <T1, T2, T3, R> combine(
    flow: Flow<T1>,
    flow2: Flow<T2>,
    flow3: Flow<T3>,
    @BuilderInference transform: suspend (T1, T2, T3) -> R
): Flow<R> = combineUnsafe(flow, flow2, flow3) { args: Array<*> ->
    transform(
        args[0] as T1,
        args[1] as T2,
        args[2] as T3
    )
}
```
Unfortunately, for more flows we have this:
```
public inline fun <reified T, R> combine(
    vararg flows: Flow<T>,
    crossinline transform: suspend (Array<T>) -> R
): Flow<R> = flow {
    combineInternal(flows, { arrayOfNulls(flows.size) }, { emit(transform(it)) })
}
```
And the result will look like that:
```
combine(
                input,
                selectedAttachments,
                lastActiveAction,
                validationErrors,
                mentionSuggestions,
                commandSuggestions,
                cooldownTimer,
) { array ->
                state.value.copy(
                    inputValue = array[0] as String,
                )
}
```

Reply:
Yeah this seems much uglier to me than the approach we have in place (unfortunately). In case we re-arrange the properties, update their types etc, we can have bugs and errors, because of the array fetching and type casting.

I think this is okay and if the number of properties start to grow, we could easily wrap them in smaller data classes and then propagate only small chunks of data.